messaging/src/main/java/org/axonframework/messaging/QualifiedName.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import jakarta.annotation.Nonnull;
+
+import static org.axonframework.common.BuilderUtils.assertNonEmpty;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Interface describing a qualified name.
+ * <p>
+ * Useful to provide clear names to {@link Message Messages}, {@link MessageHandler MessageHandlers}, and other
+ * components that require naming.
+ *
+ * @author Allard Buijze
+ * @author Mitchell Herrijgers
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public interface QualifiedName {
```

Comment:
That actually aligns with my gut feeling. I'll adjust it to a `final class`, though, because of the desire to return an `Optional<String>`. For more on that decision, please see my comment on your comment at the `revision()` method.